### PR TITLE
New Select: Fix not being able to click around the caret

### LINKED ISSUE #95980: New Select: Can't click some parts around the caret
Some parts of the Input (around the caret) don't open the menu when clicked:

https://github.com/user-attachments/assets/1b68866e-39d9-4e24-b861-f9c041e71487

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What is this feature?**

Currently because the `Input` suffix isn't clickable, there are some "blind spots" where you can click on the `Combobox` and it does not open:

https://github.com/user-attachments/assets/1b68866e-39d9-4e24-b861-f9c041e71487

This PR moves the suffix icons outside of the `suffix` prop, so you can click the box everywhere:

https://github.com/user-attachments/assets/7e3230b8-84cf-4e80-9c0d-b5efdbfec84b

**Why do we need this feature?**

So there are no "blind spots" in the new Combobox.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #95980

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-ui/src/components/Combobox/Combobox.tsx [hunk 1]
```diff
@@ -260,36 +260,9 @@ export const Combobox = <T extends string | number>({
       : 'angle-down';
 
   return (
-    <div>
+    <Box position="relative">
       <InputComponent
         width={width === 'auto' ? undefined : width}
-        suffix={
```
Comment (aocenas) [inline packages/grafana-ui/src/components/Combobox/Combobox.tsx:266]: Is this something to be fixed in the `InputComponent` itself?

Comment (tskarhed) [reply packages/grafana-ui/src/components/Combobox/Combobox.tsx:266]: I think that would be acceptable. In cases where it is not desired, the consumers could always to `event.stopPropagation()`

**Review by tskarhed — COMMENTED** [review]
I think the absolute positioning causes this issue 
image

**Comment by JoaoSilvaGrafana** [comment]
Ok tried a new approach using `pointer-events: none` instead! @tskarhed @aocenas wdyt?

This makes it even harder to fix at the `Input` level because if we set `pointer-events: none` then everywhere that sets the prefix/suffix as a button would have to manually reset that.

**Comment by eledobleefe** [comment]
Hi @JoaoSilvaGrafana!
I checked and seems to work 👌  The only nit I noticed is that the cursor does not change when I hover on the icons, keeps showing the text one. Should it change to the hand one?

**Comment by JoaoSilvaGrafana** [comment]
> Hi @JoaoSilvaGrafana! I checked and seems to work 👌 The only nit I noticed is that the cursor does not change when I hover on the icons, keeps showing the text one. Should it change to the hand one?

@eledobleefe For the clearable icon it turns into a hand pointer right? It is just the caret icon, because it has `pointer-events: none` it just takes the same hover icon as the "parent". We could set just the icon back to `pointer-events: all` and add that cursor to it, but then if you hover over the caret it would have a hand cursor, but in the pixels just around it it would still be the text one, so not sure if that is ideal...

**Review by tskarhed — APPROVED** [review]
This is en elegant solution 😄

**Comment by eledobleefe** [comment]
> > Hi @JoaoSilvaGrafana! I checked and seems to work 👌 The only nit I noticed is that the cursor does not change when I hover on the icons, keeps showing the text one. Should it change to the hand one?
> 
> @eledobleefe For the clearable icon it turns into a hand pointer right? It is just the caret icon, because it has `pointer-events: none` it just takes the same hover icon as the "parent". We could set just the icon back to `pointer-events: all` and add that cursor to it, but then if you hover over the caret it would have a hand cursor, but in the pixels just around it it would still be the text one, so not sure if that is ideal...

Yes, it keeps the text cursor when hovering on both the magnifier and the carrot icon, but it turns into pointer when hovering on the cross icon. 
🤔 Yeah, I don't really like it changing from text to pointer and viceversa within a few pixels difference.

**Review by eledobleefe — APPROVED** [review]

**Pull request merged**